libs/designer-ui/src/lib/html/plugins/toolbar/helper/Change.tsx

```diff
@@ -37,14 +37,15 @@ const convertEditorState = (editor: LexicalEditor, nodeMap: Map<string, ValueSeg
       const attributes = Array.from(element.attributes);
       for (let j = 0; j < attributes.length; j++) {
         const attribute = attributes[j];
-        if (attribute.name !== 'id' && attribute.name !== 'style') {
+        if (attribute.name !== 'id' && attribute.name !== 'style' && attribute.name !== 'href') {
           element.removeAttribute(attribute.name);
         }
       }
     }
 
     // Get the cleaned HTML string
-    const cleanedHtmlString = tempElement.innerHTML;
+    let cleanedHtmlString = tempElement.innerHTML;
+    cleanedHtmlString = cleanedHtmlString.replace(/<span>(.*?)<\/span>/g, '$1');
```

Comment:
yeah lexical adds a bunch of span tags because it does it off nodes instead of the dom elements, so when there are span tags with no attributes, I just remove them

Reply:
What if the definition already had some span without attributes, would we just update users value when they open the action because editor loaded it

Reply:
I think that’s probably fine here since span without attributes are a no op without corresponding css

Reply:
Also it should be spans that Lexical created in $generateHtmlFromNodes(editor) that were based on, I don't believe old designer would create these empty spans

Reply:
Yeah I just meant if user had them in their definition, we might start showing Save as enabled because we updated the parameter value and not the user

Reply:
something we also need to change is that save is being enabled on parameter blur instead of when a value is changed, we should compare a workflow with the previous workflow and only enable the save button if something has actually changed